adserver/views.py

```diff
@@ -641,6 +641,12 @@ def get(self, request, advertisement_id, nonce):
             advertisement.track_impression(
                 request, self.impression_type, publisher=publisher, offer=offer
             )
+        elif self.impression_type == VIEWS and "view_time" in request.GET:
```

Comment:
Should we `return` from this function? It seems like we're calling all the logic here again, which feels fraught (eg. sending to analytics, etc.) 

I worry in general about the implementation of hitting the exact same view, but moving this logic higher up in the stack and returning from it would probably reduce my worry.

Reply:
Views never send anything to analytics but I think your worry is a sound one. I'll see if I can make some changes to improve that.